hashicorp/boundary-ui
# fix: 🐛 Fix windows being unable to search client daemon

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-13466

## Description
When using the client daemon to search on windows, we use the CLI instead of the unix socket for our requests. Unfortunately that means we spawn a separate process to retrive the results and one idiosyncracy of this is that `spawnSync` buffers its outputs which means we were hitting the default max buffer size. `spawn` doesn't have this issue because it streams its outputs and we didn't previously use `spawnSync` for any command that produced large amounts of output.

The node default is 1MB and ~1000 targets will end up being around there from my quick testing. Because we weren't getting the full result set, `jsonify` would silently error out trying to parse the output and we would fall back to always calling the controller.


## How to Test
Setup boundary dev with a couple thousand targets. Either remove the if check to only use the CLI for windows or make a windows build. Confirm that we're not making any extra network requests or that `jsonify` isn't erroring out. 

## Checklist:
